Automattic/wp-calypso
# Untangling: revert Settings -> General -> Manage connection untangling

Related to p1732866860150249/1732854176.394259-slack-C06DN6QQVAQ

Reverts https://github.com/Automattic/wp-calypso/pull/96577
Resolves https://github.com/Automattic/dotcom-forge/issues/9863

## Proposed Changes

This PR moves the Settings -> General -> Manage connection back to the Calypso unification screen. This screen is used only for Jetpack sites, which don't currently open the hosting dashboard when clicked from `/sites`. We thought of moving it to the hosting dashboard because it lives in Settings -> General.

However, it turns out that Jetpack sites don't need anything in Settings -> General anymore (except this feature). See the above Slack convo for more details.

Furthermore, it was discussed that this feature should be ideally in Jetpack Cloud / wp-admin. So, we're reverting it.

|Before|After|
|-|-|
|<img width="930" alt="image" src="https://github.com/user-attachments/assets/2a07efd3-df70-4ccc-87aa-2b5398513eb9">|<img width="735" alt="image" src="https://github.com/user-attachments/assets/5c605bdd-08a9-4164-8929-9d360bdb426e">

## Why are these changes being made?

pbxlJb-6ye-p2

## Testing Instructions

1. Prepare a Jurassic Ninja site.
2. Connect it with Jetpack.
3. Go to `/settings/manage-connection/:siteSlug`
4. Verify you see the original manage connection screen as per above screenshot.

## Pre-merge Checklist
